src/App.tsx

```diff
@@ -1,26 +1,248 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, { useEffect, useRef, useState } from 'react';
+import { getTodos, postTodo, removeTodo, updateTodo } from './api/todos';
+import { Todo } from './types/Todo';
+import { TodosList } from './components/TodosList';
+import { FilterType } from './types/FilterType';
+import { Footer } from './components/Footer';
+import { ErrorType } from './types/ErrorType';
+import { ErrorNotification } from './components/ErrorNotification';
+import { MyInput } from './components/Header';
+import { OptionalAttributes } from './types/OptionalAttributes';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [error, setError] = useState<ErrorType | null>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState<FilterType>(FilterType.All);
+  const [tmpTodo, setTmpTodo] = useState<Omit<Todo, 'userId'> | null>(null);
+  const [updatedTodosId, setUpdatedTodosId] = useState<number[]>([]);
+  const [failedTitleUpdateId, setFailedTitleUpdateId] = useState<number>(0);
+
+  const downloadTodos = async () => {
+    try {
+      const tmp = await getTodos();
+
+      setTodos(tmp);
+    } catch {
+      setError(ErrorType.UnableToLoad);
+    }
+  };
+
+  useEffect(() => {
+    downloadTodos();
+    inputRef.current?.focus();
+  }, []);
+
+  const filterTodos = () => {
+    if (query === FilterType.Active) {
+      return todos.filter(todo => !todo.completed);
+    } else if (query === FilterType.Completed) {
+      return todos.filter(todo => todo.completed);
+    } else {
+      return todos;
+    }
+  };
+
+  const filteredTodos = filterTodos();
+
+  const returnLeftNumber = () => {
+    return todos.filter(todo => todo.completed != true).length;
+  };
+
+  const handleSetQuery = (passedQuery: FilterType) => {
+    setQuery(passedQuery);
+  };
+
+  const handleSetError = (errorType: ErrorType | null) => {
+    setError(errorType);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    if (inputRef.current?.value.trim()) {
+      inputRef.current.disabled = true;
+      setTmpTodo({
+        title: inputRef.current.value.trim(),
+        id: -1,
```

Comment:
why -1?

Reply:
because this todo is temporary, so I decided to provide a negative index to make it obvious